native/src/main/scala/sqs4s/SqsConsumer.scala

```diff
@@ -0,0 +1,127 @@
+package sqs4s.native
+
+import cats.effect.{Clock, Concurrent}
+import cats.implicits._
+import fs2._
+import org.http4s.client.Client
+import sqs4s.api.lo.{DeleteMessage, ReceiveMessage}
+import sqs4s.api.{ConsumerSettings, SqsSettings}
+import sqs4s.native.serialization.SqsDeserializer
+
+trait SqsConsumer[F[_], T] {
+  def consume(process: T => F[Unit]): F[Unit]
+  def consumeAsync(
+    maxConcurrent: Int,
+    batch: Int = 256
+  )(
+    process: T => F[Unit]
+  ): F[Unit]
+  def dequeue(): Stream[F, T]
+  def dequeueAsync(maxConcurrent: Int, batch: Int = 256): Stream[F, T]
+  def peek(number: Int): Stream[F, T]
+}
+
+object SqsConsumer {
+  def instance[F[_]: Concurrent: Clock: Client, T: SqsDeserializer[F, ?]](
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method is longer than 50 lines](https://app.codacy.com/manual/d2a4u/sqs4s/pullRequest?prid=4059007)